From 2924f4c473f6328ddc56bf54b6b5779d65d3b9e8 Mon Sep 17 00:00:00 2001
From: "Rafael J. Wysocki" <rjw@sisk.pl>
Date: Sat, 11 Feb 2012 22:40:23 +0100
Subject: PM / Sleep: Remove unnecessary label from suspend_freeze_processes()

The Finish label in suspend_freeze_processes() is in fact unnecessary
and makes the function look more complicated than it really is, so
remove that label (along with a few empty lines).

Signed-off-by: Rafael J. Wysocki <rjw@sisk.pl>
Acked-by: Srivatsa S. Bhat <srivatsa.bhat@linux.vnet.ibm.com>
(cherry picked from commit 6f585f750d792652f33b6e85b1ee205be4b5e572)

Signed-off-by: Simon Horman <horms@verge.net.au>
---
 kernel/power/power.h |    5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/kernel/power/power.h b/kernel/power/power.h
index 21724ee..398d42b 100644
--- a/kernel/power/power.h
+++ b/kernel/power/power.h
@@ -234,16 +234,14 @@ static inline int suspend_freeze_processes(void)
 	int error;
 
 	error = freeze_processes();
-
 	/*
 	 * freeze_processes() automatically thaws every task if freezing
 	 * fails. So we need not do anything extra upon error.
 	 */
 	if (error)
-		goto Finish;
+		return error;
 
 	error = freeze_kernel_threads();
-
 	/*
 	 * freeze_kernel_threads() thaws only kernel threads upon freezing
 	 * failure. So we have to thaw the userspace tasks ourselves.
@@ -251,7 +249,6 @@ static inline int suspend_freeze_processes(void)
 	if (error)
 		thaw_processes();
 
- Finish:
 	return error;
 }
 
-- 
1.7.10.1.362.g242cab3

